app/src/androidTest/java/de/test/antennapod/ui/PreferencesTest.java

```diff
@@ -35,36 +46,60 @@
 import static androidx.test.espresso.action.ViewActions.swipeUp;
 import static androidx.test.espresso.assertion.ViewAssertions.doesNotExist;
 import static androidx.test.espresso.assertion.ViewAssertions.matches;
+import static androidx.test.espresso.matcher.ViewMatchers.hasDescendant;
 import static androidx.test.espresso.matcher.ViewMatchers.isChecked;
 import static androidx.test.espresso.matcher.ViewMatchers.isDisplayed;
 import static androidx.test.espresso.matcher.ViewMatchers.isRoot;
 import static androidx.test.espresso.matcher.ViewMatchers.withClassName;
+import static androidx.test.espresso.matcher.ViewMatchers.withHint;
 import static androidx.test.espresso.matcher.ViewMatchers.withId;
 import static androidx.test.espresso.matcher.ViewMatchers.withText;
 import static de.test.antennapod.EspressoTestUtils.clickPreference;
 import static de.test.antennapod.EspressoTestUtils.waitForView;
 import static java.util.concurrent.TimeUnit.MILLISECONDS;
+import static org.hamcrest.Matchers.allOf;
 import static org.hamcrest.Matchers.endsWith;
 import static org.hamcrest.Matchers.not;
 import static org.junit.Assert.assertTrue;
 
 @LargeTest
 public class PreferencesTest {
     private Resources res;
+    private Activity activity;
 
     @Rule
     public ActivityTestRule<PreferenceActivity> mActivityRule = new ActivityTestRule<>(PreferenceActivity.class, false, false);
 
+
     @Before
     public void setUp() {
         EspressoTestUtils.clearDatabase();
         EspressoTestUtils.clearPreferences();
         mActivityRule.launchActivity(new Intent());
-        SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(mActivityRule.getActivity());
+        activity = mActivityRule.getActivity();
+        SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(activity);
         prefs.edit().putBoolean(UserPreferences.PREF_ENABLE_AUTODL, true).commit();
 
         res = mActivityRule.getActivity().getResources();
-        UserPreferences.init(mActivityRule.getActivity());
+        UserPreferences.init(activity);
+    }
+
+    @Test
+    public void testHideKeyboardWhenNavigatingUp() {
+        InputMethodManager imm = (InputMethodManager) mActivityRule.getActivity()
+                .getSystemService(Context.INPUT_METHOD_SERVICE);
+        onView(withId(R.id.recycler_view)).perform(
+                RecyclerViewActions.actionOnItem(
+                        allOf(hasDescendant(withHint(R.string.preference_search_hint))),
+                        click()));
+        onView(withClassName(new IsEqual<>(AppCompatImageButton.class.getName()))).perform(click());
+        Awaitility.await().atMost(1000, MILLISECONDS)
+                .until(()-> {
```

Comment:
Thanks for switching to Awaitility. That makes the test faster and more clean.

Please add a whitespace to make Checkstyle happy :)

```suggestion
                .until(() -> {
```